core/shared/src/main/scala/zio/ZQueue.scala

```diff
@@ -277,27 +277,33 @@ object ZQueue {
      * Poll all items from the queue
      */
     def unsafePollAll[A](q: MutableConcurrentQueue[A]): List[A] = {
+      val empty = null.asInstanceOf[A]
+
       @tailrec
       def poll(as: List[A]): List[A] =
-        q.poll(null.asInstanceOf[A]) match {
+        q.poll(empty) match {
           case null => as
           case a    => poll(a :: as)
         }
+
       poll(List.empty[A]).reverse
     }
 
     /**
      * Poll n items from the queue
      */
     def unsafePollN[A](q: MutableConcurrentQueue[A], max: Int): List[A] = {
+      val empty = null.asInstanceOf[A]
```

Comment:
For this one since we are polling `n` items from the queue, it may make sense to allocate an n element Array and just put them on the array, returning a view of that array later.

Reply:
Will tackle this one in a follow-up, as part of migration to chunks :).